### ISSUE TITLE
Incorrect bounding rects returned from pygame.draw.circle

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#1122]
The bounding rects returned from `pygame.draw.circle` are incorrect in some instances. The following code gives an example of an incorrect bounding rect.

```python
import pygame

def create_bounding_rect(surface, start, surf_color):
    """Create a rect to bound all the pixels that don't match surf_color."""
    width, height = surface.get_clip().size
    xmin, ymin = width, height
    xmax, ymax = -1, -1

    for y in range(height):
        for x in range(width):
            if surface.get_at((x, y)) != surf_color:
                xmin = min(x, xmin)
                xmax = max(x, xmax)
                ymin = min(y, ymin)
                ymax = max(y, ymax)

    if -1 == xmax:
        # No points means 0 sized rect with the position at the start.
        return pygame.Rect(start, (0, 0))

    return pygame.Rect((xmin, ymin), (xmax - xmin + 1, ymax - ymin + 1))


surf_color = pygame.Color('black')
surface = pygame.display.set_mode((300, 300))
surface.fill(surf_color)
pos = surface.get_rect().center

bounding_rect = pygame.draw.circle(surface, pygame.Color('red'), pos, 21, 1)
actual_rect = create_bounding_rect(surface, pos, surf_color)

print('{} bounding rect'.format(bounding_rect))
print('{} actual bounding rect'.format(actual_rect))
```

The above code gives the following output showing the actual bounding rect is different than the bounding rect returned from `pygame.draw.circle`. The circle being drawn is bigger than what is being reported as drawn.

```
pygame 2.0.0.dev2 (SDL 2.0.9, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(129, 129, 42, 42)> reported bounding rect
<rect(129, 129, 43, 42)> actual bounding rect
```

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 2.0.0.dev2 (SDL: 2.0.9) at f0de1695da4ba8ede907e09d415fce8d65f6dfc7

**Comment by charlesej** [comment]
Transferring the discussion from #1121.

There are, however, a few items that need addressing:

- The change in the way circles are being drawn (now honoring the `center` parameter) should be noted in the draw documentation and maybe add an example for clarity.
- The `test_circle__bounding_rect` test is still failing due to an assumption about what bounding rect values are returned when nothing is drawn. Currently it assumes that when nothing is drawn the returned bounding rect is (x_center, y_center, 0, 0). The draw circle code is returning (x_center, y_center, 0, 1) in the case noted above. The options here are:
   - to change the test code to accept a width and/or height of 0 to indicate nothing is drawn
   - or change the draw circle code to return a rect of (x_center, y_center, 0, 0) when nothing is drawn (my preference is this option)

**Comment by charlesej** [comment]
In addition to the `test_circle__bounding_rect` test failing as mentioned above, there is still a deficiency in how the circle draw code calculates bounding rects. In some cases where the circle is drawn off the surface (or outside the clipping area) the bounding rect is incorrect. The following code shows 2 cases. The first case draws the circle on the screen and the bounding rect is correct. In the second case the circle is drawn outside the clipping area (so nothing is drawn), but the bounding rect is reporting that something was drawn.

```python
import pygame
import time

pygame.display.init()

surface = pygame.display.set_mode((200, 200))
pos = (20, 20)
radius = 20

# This bounding rect is correct.
bounding_rect = pygame.draw.circle(surface, pygame.Color('red'), pos, radius, 1)
print('{} pre-clip bounding rect'.format(bounding_rect))

pygame.display.flip()
time.sleep(2)

# Clear the screen and clip the surface so the circle is not drawn.
surface.fill(pygame.Color('black'))
surface.set_clip(pygame.Rect(36, 36, 200, 200))

# This bounding rect is incorrect.
bounding_rect = pygame.draw.circle(surface, pygame.Color('red'), pos, radius, 1)
print('{} post-clip bounding rect'.format(bounding_rect))

pygame.display.flip()
time.sleep(2)

pygame.display.quit()
```

The following is the output from the above code.
```
pygame 2.0.0.dev3 (SDL 2.0.9, python 3.7.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(0, 0, 41, 41)> pre-clip bounding rect
<rect(36, 36, 5, 5)> post-clip bounding rect
```

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit)
- pygame: 2.0.0.dev3 (SDL: 2.0.9) at c8c9e8236cb8ad4178594badb1f29e95ff0c0708

**Comment by charlesej** [comment]
The `test_circle__bounding_rect` test was added with an `@unittest.expectedFailure` decorator via #1132.

**Comment by illume** [comment]
There's still an error left. But less of an error now :) hehe. Getting close with the circles!

**Comment by charlesej** [comment]
Resolved by #1494.
